deployment/terraform/firewall.tf

```diff
@@ -0,0 +1,67 @@
+# allow all inbound traffic to the load balancer on port 443
+resource "aws_security_group_rule" "app_alb_ingress_https" {
+  type             = "ingress"
+  from_port        = 443
+  to_port          = 443
+  protocol         = "tcp"
+  cidr_blocks      = ["0.0.0.0/0"]
+  ipv6_cidr_blocks = ["::/0"]
+
+  security_group_id = "${aws_security_group.alb.id}"
+}
+
+# allow outbound traffic from the alb to the ecs service on the app port
+resource "aws_security_group_rule" "app_alb_egress_all" {
+  type      = "egress"
+  from_port = "${var.app_port}"
+  to_port   = "${var.app_port}"
+  protocol  = "tcp"
+
+  security_group_id        = "${aws_security_group.alb.id}"
+  source_security_group_id = "${aws_security_group.ecs_service.id}"
+}
+
+# allow inbound traffic from the alb to the ecs service on the app port
+resource "aws_security_group_rule" "app_ecs_cluster_ingress_alb" {
```

Comment:
The naming of this resource is slightly inconsistent with the ones above it. `app` is being used to represent the security group associated with the ECS service, but here it is `app_ecs_cluster`. That makes it seem like the contents of the `aws_security_group_rule` is referencing different entities, but they are the same (`aws_security_group.ecs_service.id`).

I think continuing to just use `app` makes the most sense.

Reply:
I thought I did a find and replace on `ecs_cluster` with `ecs_service`, but I must have missed the security group rule names. Thank you for catching this.

Reply:
I agree that it makes more sense to just refer to this security group by `app`.